sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlockBlobAsyncClient.java

```diff
@@ -379,19 +377,12 @@ private String getBlockID() {
      *
      * @return A reactive response containing the list of blocks.
      */
-    public Flux<BlockItem> listBlocks(BlockListType listType,
+    public Mono<Response<BlockList>> listBlocks(BlockListType listType,
```

Comment:
Did we have a conversation with Jonathan about non-paged listing operations returning Mono<List>? It seems to me that Flux is more natural, but if that discussion has been had, then I'll go along with it.

Reply:
@JonathanGiles has there been a decision on return types for listing operations that aren't paginated?

Reply:
Tracked separately under #5097

Reply:
Yeah - this change isn't what I expected to see either. I would have preferred to see a PagedFlux here, even though there is only one page. I'll clarify the spec on this this week. I also suspect you would like to see improvements to the PagedFlux constructors for situations where there is no follow-up pages in the response - please do feel free to submit PRs to improve that (or else let me know and I will ensure it gets improved).